staging/src/k8s.io/apiserver/pkg/storage/cacher/delegator.go

```diff
@@ -139,6 +139,10 @@ func (c *CacheDelegator) Get(ctx context.Context, key string, opts storage.GetOp
 }
 
 func (c *CacheDelegator) GetList(ctx context.Context, key string, opts storage.ListOptions, listObj runtime.Object) error {
+	_, _, err := storage.ValidateListOptions(c.cacher.resourcePrefix, c.cacher.versioner, opts)
```

Comment:
This is a functional change - we start validating something that we didn't validate before.

Can you please move it to a separate PR, as we need a careful review for this part (whether we don't break something from user POV).

Reply:
Technically the validation doesn't change just the flow changes. Flows:
* Previous: Delegate.GetList -> shouldDelegate -> etcd.GetList -> validate via resolveGetListRev
* New: Delegate.GetList -> validate via ValidateListOptions

Reply:
The question is what happens if we actually don't delegate to etcd...
Let's discuss in a follow-up PR.